### ISSUE TITLE
match-equals doesn't work correctly with absent in 3.1.1.

### ISSUE THREAD

**Issue by eduardoeof** [nubank/matcher-combinators#151]
We bumped the match-combinator's version of a service and a test start to break for no reason. That test was using match-equals and absent. The version bump was from 1.2.5 to 3.1.1.

**Comment by philomates** [comment]
to reproduce:
```
(ns matcher-combinators.scratch
  (:require [clojure.test :refer :all]
            [matcher-combinators.test :refer :all]
            [matcher-combinators.core :as core]
            [matcher-combinators.matchers :as matchers]))

(deftest match-equals-with-absent
  (testing "absent"
    (is (match? (matchers/match-with [map? matchers/equals]
                                     {:a matchers/absent})
                {:b 1}))))
```

```
expected: (match? (matchers/match-with [map? matchers/equals] {:a matchers/absent}) {:b 1})
  actual: java.lang.ClassCastException: class matcher_combinators.core.Absent cannot be cast to class clojure.lang.IFn (matcher_combinators.core.Absent is in unnamed module of l
oader clojure.lang.DynamicClassLoader @49f71bea; clojure.lang.IFn is in unnamed module of loader 'app')
 at matcher_combinators.matchers$match_with.invokeStatic (matchers.cljc:169)
    matcher_combinators.matchers$match_with.invoke (matchers.cljc:132)
    matcher_combinators.matchers$match_with_values$fn__19275.invoke (matchers.cljc:121)
    clojure.lang.PersistentArrayMap.kvreduce (PersistentArrayMap.java:377)
    clojure.core$fn__8437.invokeStatic (core.clj:6845)
    clojure.core/fn (core.clj:6830)
    clojure.core.protocols$fn__8167$G__8162__8176.invoke (protocols.clj:175)
    clojure.core$reduce_kv.invokeStatic (core.clj:6856)
    clojure.core$reduce_kv.invoke (core.clj:6847)
    matcher_combinators.matchers$match_with_values.invokeStatic (matchers.cljc:121)
    matcher_combinators.matchers$match_with_values.invoke (matchers.cljc:120)
    matcher_combinators.matchers$match_with.invokeStatic (matchers.cljc:170)

```

**Comment by philomates** [comment]
should be fixed in `3.1.3`
